[now-cli][now-client] (Major) Remove legacy code

AndyBitz opened the pull request:
Removes legacy and unused code.

Fixes: https://sentry.io/organizations/zeithq/issues/1524784141/?project=1323225

[PRODUCT-1213]

[PRODUCT-1213]: https://zeit.atlassian.net/browse/PRODUCT-1213

AndyBitz committed:
[now-cli][now-client] Remove legacy code

AndyBitz committed:
Adjust tests

AndyBitz committed:
Merge branch 'master' of https://github.com/zeit/now into now-cli/remove-legacy-code

AndyBitz committed:
Adjust more tests

styfle commented on packages/now-cli/package.json:
```diff
@@ -130,8 +127,6 @@
     "get-port": "5.1.1",
     "glob": "7.1.2",
     "http-proxy": "1.17.0",
-    "ignore": "4.0.6",
```
I'm surprised, I thought we used `ignore` in v2 also 🤔

AndyBitz replied:
@styfle yes, but it's in `now-client`

TooTallNate replied:
I think it's used by `now dev` as well? 🤔

styfle commented on packages/now-cli/src/commands/alias/set.ts:
```diff
@@ -284,34 +185,6 @@ function handleSetupDomainError<T>(
         { extraSpace: '     ' }
       )}\n\n`
     );
-    if (error instanceof ERRORS.DomainVerificationFailed && !isWildcard) {
```
Is this error only v1 or can it happen in v2 too?

AndyBitz replied:
We still check that error, but we don't make it depend on the `isWildcard` variable

styfle commented on packages/now-cli/src/commands/certs/ls.ts:
```diff
@@ -130,13 +138,18 @@ function formatCertCn(cn: string, multiple: boolean) {
     formatExpirationDate(new Date(cert.expiration)),
     cert.autoRenew ? 'yes' : 'no',
-    chalk.gray(ms(time.getTime() - new Date(cert.created).getTime()))
+    chalk.gray(ms(time.getTime() - new Date(cert.created).getTime())),
```
Nice catch with `ms`! Should this also say "ago" like the `rm` command?

styfle replied:
I guess rm is using Date.now() but this one is given a time variable so maybe not 🤔

rdev commented on packages/now-client/src/create-deployment.ts:
```diff
@@ -1,205 +1,170 @@
   DeploymentEventType,
 } from './types';
 
-export default function buildCreateDeployment(version: number) {
```
Should we detect v1 and yield an `error`? 🤔

AndyBitz replied:
Nope, everything 1.0 related is gone.

styfle commented on packages/now-cli/src/commands/deploy/index.js:
```diff
@@ -112,89 +84,5 @@ export default async ctx => {
-          const first = code(1);
-          const second = code(2);
-
-          output.error(
```
I think we might want to keep an error here when `typeof version === 'number' && version !== 2` so that we can fail if someone tries `version: 3` for example.

styfle replied:
And maybe we can print a helpful error if they have ` version: 1` that says its no longer supported and links to the upgrade docs.

AndyBitz replied:
The API would return an error in that case. Ideally we should remove all the validation from Now CLI and let the API handle everything => less duplicated code, no new version when updating error messages etc.

styfle commented on packages/now-cli/src/commands/index.ts:
```diff
@@ -6,8 +6,6 @@ export default new Map([
-  ['deploy-v2', 'deploy'],
   ['dev', 'dev'],
   ['dns', 'dns'],
   ['domain', 'domains'],
```
Can we also remove `scale`?

styfle commented on packages/now-cli/src/commands/inspect.js:
```diff
@@ -70,7 +65,7 @@ export default async function main(ctx) {
 
   // extract the first parameter
-  id = argv._[1];
+  const [deploymentIdOrHost] = argv._;
```
Are you sure this is right? It was using index 1 before? So I would have expected something like:

```suggestion
  const [_, deploymentIdOrHost] = argv._;
```

styfle commented on packages/now-cli/src/util/alias/assign-alias.ts:
```diff
@@ -21,71 +12,11 @@ export default async function assignAlias(
-  // deployment, we need to account for it here.
-  if (prevDeployment instanceof ERRORS.DeploymentNotFound) {
-    prevDeployment = null;
-  }
```
Are you sure this won't return DeploymentNotFound error for v2?

What about it `prevDeployment` is an error?

AndyBitz replied:
`prevDeployment` was only used for `now alias` without any arguments, which has been deprecated a while ago. So there is no such thing as a previous deployment anymore.

styfle commented on packages/now-cli/src/util/client.ts:
```diff
@@ -63,19 +63,19 @@ export default class Client extends EventEmitter {
 
     if (opts.accountId || opts.useCurrentTeam !== false) {
-      const query = parsedUrl.query;
+      const query = new URLSearchParams(parsedUrl.query);
```
I think there are some small differences between `querystring` and `URLSearchParams` but this will probably be fine for our purposes.

styfle replied:
@AndyBitz I just realized you can probably use `new URL()` here too instead of `parseUrl(_url, true)`

AndyBitz replied:
Not in this case unfortunately, because `new URL()` requires a complete URL (including host and protocol), `url.parse` works with just a pathname.

AndyBitz committed:
Remove scale command and --no-scale option from help

AndyBitz committed:
Update packages/now-cli/src/commands/inspect.js

Co-Authored-By: Steven <steven@ceriously.com>

styfle commented on packages/now-cli/src/util/get-files.ts:
```diff
@@ -183,7 +161,7 @@ export async function staticFiles(
 
     // Compile list of ignored patterns and files
     const ignoreName = isBuilds ? '.nowignore' : '.gitignore';
```
I think `isBuilds` is always true since the comment says `true for Now 2.0 builders`. So can we remove some more logic in this function?

AndyBitz replied:
Good catch 👍 
Removed it

styfle commented on packages/now-cli/src/util/ignored.ts:
```diff
@@ -1,25 +0,0 @@
-node_modules
-__pycache__/
-venv/
-CVS`;
```
This is great because we had a duplicate it `now-client` so now there is one source of truth! 🎉

AndyBitz replied:
Exactly, there is probably even more that can be unified, e.g. the getFiles function that is used in `now-dev`

styfle commented on packages/now-cli/test/helpers/prepare.js:
```diff
@@ -469,36 +297,28 @@ CMD ["node", "index.js"]`,
-        await writeFile(file.replace('-builds', ''), content);
-      }
+    // Get content from the object property
+    for (const name of Object.keys(needed)) {
```
```suggestion
    for (const [name, content] of Object.entries(needed)) {
```

AndyBitz committed:
Remove isBuilds option

AndyBitz committed:
Use Object.entries

styfle commented on packages/now-cli/test/integration.js:
```diff
@@ -753,47 +670,19 @@ test('set platform version using `-V` to `2`', async t => {
 });
 
 test('output logs of a 2.0 deployment without annotate', async t => {
+  const deployment = await createDeploymentWithFixture();
```
This might make our tests a little slower since its creating a deployment for each test. Previously, this `logs` test was reading the logs of the previous deployment.

AndyBitz replied:
@styfle I am aware, but ideally the deployment should be deduped, so it's not that much overhead. And I want every test to work on it's own, so that we can remove the `context` stuff and use `test.only` without running into any issues because of that.

styfle commented on packages/now-cli/test/unit.js:
```diff
@@ -810,50 +240,6 @@ test('getProjectName with a directory', t => {
-  const res = await fetch(url);
-  const formatted = await responseErrorMessage(res);
-
-  t.is(formatted, 'Response Error (500)');
```
Why remove the response error tests here?

AndyBitz replied:
The whole function is not used anywhere in the code

styfle commented on packages/now-client/package.json:
```diff
@@ -45,7 +45,7 @@
     "ms": "2.1.2",
     "node-fetch": "2.6.0",
-    "querystring": "^0.2.0",
+    "rambda": "4.5.0",
```
Is `rambda` necessary?

AndyBitz replied:
Nope, replaced it

styfle commented on packages/now-client/src/types.ts:
```diff
@@ -95,13 +95,10 @@ export interface DeploymentGithubData {
-
-export interface NowConfig extends LegacyNowConfig {
+export interface NowConfig {
+  [key: string]: any;
```
Is there a reason why `[key: string]: any;` was added here? Can we add any additional properties instead?

AndyBitz replied:
Had it temporarily to fix something, but removed it now 👍 
and even more

AndyBitz committed:
Remove rambda

AndyBitz committed:
Adjust type and remove more unused code

styfle commented on packages/now-client/src/utils/index.ts:
```diff
@@ -140,10 +135,10 @@ export const fetch = async (
 
   if (opts.teamId) {
     const parsedUrl = parseUrl(url, true);
```
You could replace `parseUrl(url, true)` with `new URL(url)` I believe.

AndyBitz replied:
Good point

styfle commented on packages/now-client/src/utils/index.ts:
```diff
@@ -36,12 +35,8 @@ export type DeploymentEventType = (typeof EVENTS_ARRAY)[number];
-  }
-
   if (metadata && metadata.builds && !metadata.functions) {
     return '/v10/now/deployments';
```
Should this be `v12`?

AndyBitz replied:
Nope, 2 lines below there is the `v12` route option, but for `builds` we need to use `v10`

AndyBitz committed:
Use URL instead of url.parse

AndyBitz committed:
Fix linting

styfle commented on packages/now-cli/src/util/error.ts:
```diff
@@ -25,19 +42,16 @@ export async function responseError(
   }
 
-  if (message == null) {
+  if (!message) {
```
Can this whole if statement be simplified?

```js
const message = bodyError.message || fallbackMessage || 'Response Error';
```

Or do we want an error message with an empty string in some cases?

AndyBitz replied:
No, I don't think that's desirable

styfle commented on packages/now-cli/src/util/error.ts:
```diff
@@ -25,19 +42,16 @@ export async function responseError(
+  const err = new ResponseError(message, res.status);
 
   // Copy every field that was added manually to the error
   if (bodyError) {
```
You can remove `if (bodyError)` since its always true now.

styfle commented on packages/now-client/src/create-deployment.ts:
```diff
@@ -1,205 +1,173 @@
+    delete include.scope;
+    delete include.github;
+
+    Object.assign(deploymentOptions, include);
```
Nit pick: you can avoid looping twice and assign in one loop:

```ts
for (const [key, value] of Object.entries(nowConfig)) {
  if (key !== 'scope' && key !== 'github') {
    deploymentOptions[key] = value;
  }
}
```

AndyBitz replied:
Yup, TS complains

styfle reviewed (approved):
I have a couple additional comments but those are mostly nit picks.

Great work @AndyBitz! 🎉 🥇 🚀

AndyBitz committed:
Simplify responseError function

AndyBitz committed:
Merge branch 'master' into now-cli/remove-legacy-code

AndyBitz committed:
Remove GNU bash, version 5.0.11(1)-release (x86_64-apple-darwin19.0.0)
These shell commands are defined internally.  Type `help' to see this list.
Type `help name' to find out more about the function `name'.
Use `info bash' to find out more about the shell in general.
Use `man -k' or `info' to find out more about commands not in this list.

A star (*) next to a name means that the command is disabled.

 job_spec [&]                                                                             history [-c] [-d offset] [n] or history -anrw [filename] or history -ps arg [arg...]
 (( expression ))                                                                         if COMMANDS; then COMMANDS; [ elif COMMANDS; then COMMANDS; ]... [ else COMMANDS; ] >
 . filename [arguments]                                                                   jobs [-lnprs] [jobspec ...] or jobs -x command [args]
 :                                                                                        kill [-s sigspec | -n signum | -sigspec] pid | jobspec ... or kill -l [sigspec]
 [ arg... ]                                                                               let arg [arg ...]
 [[ expression ]]                                                                         local [option] name[=value] ...
 alias [-p] [name[=value] ... ]                                                           logout [n]
 bg [job_spec ...]                                                                        mapfile [-d delim] [-n count] [-O origin] [-s count] [-t] [-u fd] [-C callback] [-c q>
 bind [-lpsvPSVX] [-m keymap] [-f filename] [-q name] [-u name] [-r keyseq] [-x keyseq:>  popd [-n] [+N | -N]
 break [n]                                                                                printf [-v var] format [arguments]
 builtin [shell-builtin [arg ...]]                                                        pushd [-n] [+N | -N | dir]
 caller [expr]                                                                            pwd [-LP]
 case WORD in [PATTERN [| PATTERN]...) COMMANDS ;;]... esac                               read [-ers] [-a array] [-d delim] [-i text] [-n nchars] [-N nchars] [-p prompt] [-t t>
 cd [-L|[-P [-e]] [-@]] [dir]                                                             readarray [-d delim] [-n count] [-O origin] [-s count] [-t] [-u fd] [-C callback] [-c>
 command [-pVv] command [arg ...]                                                         readonly [-aAf] [name[=value] ...] or readonly -p
 compgen [-abcdefgjksuv] [-o option] [-A action] [-G globpat] [-W wordlist]  [-F functi>  return [n]
 complete [-abcdefgjksuv] [-pr] [-DEI] [-o option] [-A action] [-G globpat] [-W wordlis>  select NAME [in WORDS ... ;] do COMMANDS; done
 compopt [-o|+o option] [-DEI] [name ...]                                                 set [-abefhkmnptuvxBCHP] [-o option-name] [--] [arg ...]
 continue [n]                                                                             shift [n]
 coproc [NAME] command [redirections]                                                     shopt [-pqsu] [-o] [optname ...]
 declare [-aAfFgilnrtux] [-p] [name[=value] ...]                                          source filename [arguments]
 dirs [-clpv] [+N] [-N]                                                                   suspend [-f]
 disown [-h] [-ar] [jobspec ... | pid ...]                                                test [expr]
 echo [-neE] [arg ...]                                                                    time [-p] pipeline
 enable [-a] [-dnps] [-f filename] [name ...]                                             times
 eval [arg ...]                                                                           trap [-lp] [[arg] signal_spec ...]
 exec [-cl] [-a name] [command [arguments ...]] [redirection ...]                         true
 exit [n]                                                                                 type [-afptP] name [name ...]
 export [-fn] [name[=value] ...] or export -p                                             typeset [-aAfFgilnrtux] [-p] name[=value] ...
 false                                                                                    ulimit [-SHabcdefiklmnpqrstuvxPT] [limit]
 fc [-e ename] [-lnr] [first] [last] or fc -s [pat=rep] [command]                         umask [-p] [-S] [mode]
 fg [job_spec]                                                                            unalias [-a] name [name ...]
 for NAME [in WORDS ... ] ; do COMMANDS; done                                             unset [-f] [-v] [-n] [name ...]
 for (( exp1; exp2; exp3 )); do COMMANDS; done                                            until COMMANDS; do COMMANDS; done
 function name { COMMANDS ; } or name () { COMMANDS ; }                                   variables - Names and meanings of some shell variables
 getopts optstring name [arg]                                                             wait [-fn] [id ...]
 hash [-lr] [-p pathname] [-dt] [name ...]                                                while COMMANDS; do COMMANDS; done
 help [-dms] [pattern ...]                                                                { COMMANDS ; } from subcommand list since it's a global command

AndyBitz committed:
Merge branch 'master' of https://github.com/zeit/now into now-cli/remove-legacy-code

AndyBitz committed:
Log errors

AndyBitz committed:
Merge branch 'master' into now-cli/remove-legacy-code

AndyBitz committed:
Merge branch 'master' into now-cli/remove-legacy-code

pranaygp commented:
# [Codecov](https://codecov.io/gh/zeit/now/pull/3840?src=pr&el=h1) Report
> Merging [#3840](https://codecov.io/gh/zeit/now/pull/3840?src=pr&el=desc) into [master](https://codecov.io/gh/zeit/now/commit/095805e3ad09c277e4aa0a8cc04e0cf672b0c746&el=desc) will **increase** coverage by `0.57%`.
> The diff coverage is `n/a`.

[![Impacted file tree graph](https://codecov.io/gh/zeit/now/pull/3840/graphs/tree.svg?width=650&height=150&src=pr&token=8QQ5cmij1O)](https://codecov.io/gh/zeit/now/pull/3840?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master    #3840      +/-   ##
==========================================
+ Coverage   15.40%   15.97%   +0.57%     
==========================================
  Files         277      227      -50     
  Lines       10292     8528    -1764     
  Branches     1472     1236     -236     
==========================================
- Hits         1585     1362     -223     
+ Misses       8520     6979    -1541     
  Partials      187      187              
```


| [Impacted Files](https://codecov.io/gh/zeit/now/pull/3840?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [src/util/projects/get-project-by-id-or-name.ts](https://codecov.io/gh/zeit/now/pull/3840/diff?src=pr&el=tree#diff-c3JjL3V0aWwvcHJvamVjdHMvZ2V0LXByb2plY3QtYnktaWQtb3ItbmFtZS50cw==) | `0.00% <0.00%> (-33.34%)` | :arrow_down: |
| [src/util/output/cmd.ts](https://codecov.io/gh/zeit/now/pull/3840/diff?src=pr&el=tree#diff-c3JjL3V0aWwvb3V0cHV0L2NtZC50cw==) | `66.66% <0.00%> (-16.67%)` | :arrow_down: |
| [src/util/output/highlight.ts](https://codecov.io/gh/zeit/now/pull/3840/diff?src=pr&el=tree#diff-c3JjL3V0aWwvb3V0cHV0L2hpZ2hsaWdodC50cw==) | `66.66% <0.00%> (-16.67%)` | :arrow_down: |
| [src/util/config/local-path.ts](https://codecov.io/gh/zeit/now/pull/3840/diff?src=pr&el=tree#diff-c3JjL3V0aWwvY29uZmlnL2xvY2FsLXBhdGgudHM=) | `66.66% <0.00%> (-13.34%)` | :arrow_down: |
| [src/util/errors.ts](https://codecov.io/gh/zeit/now/pull/3840/diff?src=pr&el=tree#diff-c3JjL3V0aWwvZXJyb3JzLnRz) | `45.29% <0.00%> (-9.26%)` | :arrow_down: |
| [src/util/get-files.ts](https://codecov.io/gh/zeit/now/pull/3840/diff?src=pr&el=tree#diff-c3JjL3V0aWwvZ2V0LWZpbGVzLnRz) | `82.85% <0.00%> (-7.33%)` | :arrow_down: |
| [src/util/output/create-output.ts](https://codecov.io/gh/zeit/now/pull/3840/diff?src=pr&el=tree#diff-c3JjL3V0aWwvb3V0cHV0L2NyZWF0ZS1vdXRwdXQudHM=) | `54.71% <0.00%> (-5.67%)` | :arrow_down: |
| [src/util/secrets.js](https://codecov.io/gh/zeit/now/pull/3840/diff?src=pr&el=tree#diff-c3JjL3V0aWwvc2VjcmV0cy5qcw==) | `0.00% <0.00%> (ø)` | :arrow_up: |
| [src/commands/list.js](https://codecov.io/gh/zeit/now/pull/3840/diff?src=pr&el=tree#diff-c3JjL2NvbW1hbmRzL2xpc3QuanM=) | `0.00% <0.00%> (ø)` | :arrow_up: |
| [src/util/dev/server.ts](https://codecov.io/gh/zeit/now/pull/3840/diff?src=pr&el=tree#diff-c3JjL3V0aWwvZGV2L3NlcnZlci50cw==) | `56.83% <0.00%> (ø)` | :arrow_up: |
| ... and [80 more](https://codecov.io/gh/zeit/now/pull/3840/diff?src=pr&el=tree-more) | |

------

[Continue to review full report at Codecov](https://codecov.io/gh/zeit/now/pull/3840?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/zeit/now/pull/3840?src=pr&el=footer). Last update [995aa6e...7aee542](https://codecov.io/gh/zeit/now/pull/3840?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

AndyBitz committed:
Merge branch 'master' into now-cli/remove-legacy-code
